sklearn/utils/multiclass.py

```diff
@@ -155,6 +155,18 @@ def is_multilabel(y):
                                     _is_integral_float(labels))
 
 
+def is_onehot(y):
+    """check if y is an one-hot encoded matrix.
+    """
+    if not is_multilabel(y):
+        return False
+    elif not len(np.unique(y)) == 2:
```

Comment:
is_multilabel alrready checks this.

Reply:
function removed.

Reply:
onehot check was intended to avoid change of behaviour. although not supported, one-hot encoded matrix right now can be accepted into log_loss and calculate multi-class log_loss. without one-hot check, there is change of behaviour for such unsupported use case.

Reply:
If we want to care about this case, we need to be consistent across all metrics. Since that it was never supported, I would say that this should not be a trouble but maybe the nicest way would be to raise an error across all metrics.